gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/ManifestBasedDataset.java

```diff
@@ -78,37 +81,43 @@ public Iterator<FileSet<CopyEntity>> getFileSetIterator(FileSystem targetFs, Cop
           + "%s, you can specify multi locations split by '',", manifestPath.toString(), fs.getUri().toString(), ManifestBasedDatasetFinder.MANIFEST_LOCATION));
     }
     CopyManifest.CopyableUnitIterator manifests = null;
-    List<CopyEntity> copyEntities = Lists.newArrayList();
-    List<FileStatus> toDelete = Lists.newArrayList();
+    List<CopyEntity> copyEntities = Collections.synchronizedList(Lists.newArrayList());
+    List<FileStatus> toDelete = Collections.synchronizedList(Lists.newArrayList());
     //todo: put permission preserve logic here?
     try {
+      long startTime = System.currentTimeMillis();
       manifests = CopyManifest.getReadIterator(this.fs, this.manifestPath);
+      Cache<String, OwnerAndPermission> permissionMap = CacheBuilder.newBuilder().expireAfterAccess(30, TimeUnit.SECONDS).build();
```

Comment:
how did you arrive at 30s TTL?  seems short... I'd think more in the neighborhood of 5-15 mins

Reply:
TTL means if we haven't tried to get it within the 30s. I tested for 30k files, we take around 30 sec to plan. Also, given the truth or assumption that files share the same parent usually sit near each other in the manifest file, I think 30 sec should be enough for us. Also, this will control the memory we are using here.

Reply:
Could we make this configurable?